---
title: "exceptions"
description: "module requests.exceptions in requests"
---

## Description

    requests.exceptions
    ~~~~~~~~~~~~~~~~~~~
    
    This module contains the set of Requests' exceptions.

## Classes

    builtins.OSError(builtins.Exception)
        RequestException
            ChunkedEncodingError
            ConnectionError
                ConnectTimeout(ConnectionError, Timeout)
                ProxyError
                SSLError
            ContentDecodingError(RequestException, urllib3.exceptions.HTTPError)
            HTTPError
            InvalidHeader(RequestException, builtins.ValueError)
            InvalidJSONError
                JSONDecodeError(InvalidJSONError, json.decoder.JSONDecodeError)
            InvalidSchema(RequestException, builtins.ValueError)
            InvalidURL(RequestException, builtins.ValueError)
                InvalidProxyURL
            MissingSchema(RequestException, builtins.ValueError)
            RetryError
            StreamConsumedError(RequestException, builtins.TypeError)
            Timeout
                ReadTimeout
            TooManyRedirects
            URLRequired
            UnrewindableBodyError
    builtins.Warning(builtins.Exception)
        RequestsWarning
            FileModeWarning(RequestsWarning, builtins.DeprecationWarning)
            RequestsDependencyWarning
    
    
### ChunkedEncodingError
    
<strong>class ChunkedEncodingError(RequestException)</strong>
    
     |  ChunkedEncodingError(*args, **kwargs)
     |  
     |  The server declared chunked encoding but sent an invalid chunk.
     |  
     |  Method resolution order:
     |      ChunkedEncodingError
     |      RequestException
     |      builtins.OSError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### ConnectTimeout
    
<strong>class ConnectTimeout(ConnectionError, Timeout)</strong>
    
     |  ConnectTimeout(*args, **kwargs)
     |  
     |  The request timed out while trying to connect to the remote server.
     |  
     |  Requests that produced this error are safe to retry.
     |  
     |  Method resolution order:
     |      ConnectTimeout
     |      ConnectionError
     |      Timeout
     |      RequestException
     |      builtins.OSError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### ConnectionError
    
<strong>class ConnectionError(RequestException)</strong>
    
     |  ConnectionError(*args, **kwargs)
     |  
     |  A Connection error occurred.
     |  
     |  Method resolution order:
     |      ConnectionError
     |      RequestException
     |      builtins.OSError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### ContentDecodingError
    
<strong>class ContentDecodingError(RequestException, urllib3.exceptions.HTTPError)</strong>
    
     |  ContentDecodingError(*args, **kwargs)
     |  
     |  Failed to decode response content.
     |  
     |  Method resolution order:
     |      ContentDecodingError
     |      RequestException
     |      builtins.OSError
     |      urllib3.exceptions.HTTPError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### FileModeWarning
    
<strong>class FileModeWarning(RequestsWarning, builtins.DeprecationWarning)</strong>
    
     |  A file was opened in text mode, but Requests determined its binary length.
     |  
     |  Method resolution order:
     |      FileModeWarning
     |      RequestsWarning
     |      builtins.DeprecationWarning
     |      builtins.Warning
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Data descriptors inherited from RequestsWarning:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.DeprecationWarning:
     |  
     |  
#### DeprecationWarning.__init__
     |  
     |  __init__(self, /, *args, **kwargs)
     |      Initialize self.  See help(type(self)) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.DeprecationWarning:
     |  
     |  
#### DeprecationWarning.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### HTTPError
    
<strong>class HTTPError(RequestException)</strong>
    
     |  HTTPError(*args, **kwargs)
     |  
     |  An HTTP error occurred.
     |  
     |  Method resolution order:
     |      HTTPError
     |      RequestException
     |      builtins.OSError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### InvalidHeader
    
<strong>class InvalidHeader(RequestException, builtins.ValueError)</strong>
    
     |  InvalidHeader(*args, **kwargs)
     |  
     |  The header value provided was somehow invalid.
     |  
     |  Method resolution order:
     |      InvalidHeader
     |      RequestException
     |      builtins.OSError
     |      builtins.ValueError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### InvalidJSONError
    
<strong>class InvalidJSONError(RequestException)</strong>
    
     |  InvalidJSONError(*args, **kwargs)
     |  
     |  A JSON error occurred.
     |  
     |  Method resolution order:
     |      InvalidJSONError
     |      RequestException
     |      builtins.OSError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### InvalidProxyURL
    
<strong>class InvalidProxyURL(InvalidURL)</strong>
    
     |  InvalidProxyURL(*args, **kwargs)
     |  
     |  The proxy URL provided is invalid.
     |  
     |  Method resolution order:
     |      InvalidProxyURL
     |      InvalidURL
     |      RequestException
     |      builtins.OSError
     |      builtins.ValueError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### InvalidSchema
    
<strong>class InvalidSchema(RequestException, builtins.ValueError)</strong>
    
     |  InvalidSchema(*args, **kwargs)
     |  
     |  The URL scheme provided is either invalid or unsupported.
     |  
     |  Method resolution order:
     |      InvalidSchema
     |      RequestException
     |      builtins.OSError
     |      builtins.ValueError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### InvalidURL
    
<strong>class InvalidURL(RequestException, builtins.ValueError)</strong>
    
     |  InvalidURL(*args, **kwargs)
     |  
     |  The URL provided was somehow invalid.
     |  
     |  Method resolution order:
     |      InvalidURL
     |      RequestException
     |      builtins.OSError
     |      builtins.ValueError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### JSONDecodeError
    
<strong>class JSONDecodeError(InvalidJSONError, json.decoder.JSONDecodeError)</strong>
    
     |  JSONDecodeError(*args, **kwargs)
     |  
     |  Couldn't decode the text into json
     |  
     |  Method resolution order:
     |      JSONDecodeError
     |      InvalidJSONError
     |      RequestException
     |      builtins.OSError
     |      json.decoder.JSONDecodeError
     |      builtins.ValueError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods defined here:
     |  
     |  
#### JSONDecodeError.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Construct the JSONDecodeError instance first with all
     |      args. Then use it's args to construct the IOError so that
     |      the json specific args aren't used as IOError specific args
     |      and the error message from JSONDecodeError is preserved.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### MissingSchema
    
<strong>class MissingSchema(RequestException, builtins.ValueError)</strong>
    
     |  MissingSchema(*args, **kwargs)
     |  
     |  The URL scheme (e.g. http or https) is missing.
     |  
     |  Method resolution order:
     |      MissingSchema
     |      RequestException
     |      builtins.OSError
     |      builtins.ValueError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### ProxyError
    
<strong>class ProxyError(ConnectionError)</strong>
    
     |  ProxyError(*args, **kwargs)
     |  
     |  A proxy error occurred.
     |  
     |  Method resolution order:
     |      ProxyError
     |      ConnectionError
     |      RequestException
     |      builtins.OSError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### ReadTimeout
    
<strong>class ReadTimeout(Timeout)</strong>
    
     |  ReadTimeout(*args, **kwargs)
     |  
     |  The server did not send any data in the allotted amount of time.
     |  
     |  Method resolution order:
     |      ReadTimeout
     |      Timeout
     |      RequestException
     |      builtins.OSError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### RequestException
    
<strong>class RequestException(builtins.OSError)</strong>
    
     |  RequestException(*args, **kwargs)
     |  
     |  There was an ambiguous exception that occurred while handling your
     |  request.
     |  
     |  Method resolution order:
     |      RequestException
     |      builtins.OSError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods defined here:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors defined here:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### RequestsDependencyWarning
    
<strong>class RequestsDependencyWarning(RequestsWarning)</strong>
    
     |  An imported dependency doesn't match the expected version range.
     |  
     |  Method resolution order:
     |      RequestsDependencyWarning
     |      RequestsWarning
     |      builtins.Warning
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Data descriptors inherited from RequestsWarning:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.Warning:
     |  
     |  
#### Warning.__init__
     |  
     |  __init__(self, /, *args, **kwargs)
     |      Initialize self.  See help(type(self)) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.Warning:
     |  
     |  
#### Warning.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### RequestsWarning
    
<strong>class RequestsWarning(builtins.Warning)</strong>
    
     |  Base warning for Requests.
     |  
     |  Method resolution order:
     |      RequestsWarning
     |      builtins.Warning
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Data descriptors defined here:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.Warning:
     |  
     |  
#### Warning.__init__
     |  
     |  __init__(self, /, *args, **kwargs)
     |      Initialize self.  See help(type(self)) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.Warning:
     |  
     |  
#### Warning.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### RetryError
    
<strong>class RetryError(RequestException)</strong>
    
     |  RetryError(*args, **kwargs)
     |  
     |  Custom retries logic failed
     |  
     |  Method resolution order:
     |      RetryError
     |      RequestException
     |      builtins.OSError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### SSLError
    
<strong>class SSLError(ConnectionError)</strong>
    
     |  SSLError(*args, **kwargs)
     |  
     |  An SSL error occurred.
     |  
     |  Method resolution order:
     |      SSLError
     |      ConnectionError
     |      RequestException
     |      builtins.OSError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### StreamConsumedError
    
<strong>class StreamConsumedError(RequestException, builtins.TypeError)</strong>
    
     |  StreamConsumedError(*args, **kwargs)
     |  
     |  The content for this response was already consumed.
     |  
     |  Method resolution order:
     |      StreamConsumedError
     |      RequestException
     |      builtins.OSError
     |      builtins.TypeError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### Timeout
    
<strong>class Timeout(RequestException)</strong>
    
     |  Timeout(*args, **kwargs)
     |  
     |  The request timed out.
     |  
     |  Catching this error will catch both
     |  :exc:`~requests.exceptions.ConnectTimeout` and
     |  :exc:`~requests.exceptions.ReadTimeout` errors.
     |  
     |  Method resolution order:
     |      Timeout
     |      RequestException
     |      builtins.OSError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### TooManyRedirects
    
<strong>class TooManyRedirects(RequestException)</strong>
    
     |  TooManyRedirects(*args, **kwargs)
     |  
     |  Too many redirects.
     |  
     |  Method resolution order:
     |      TooManyRedirects
     |      RequestException
     |      builtins.OSError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### URLRequired
    
<strong>class URLRequired(RequestException)</strong>
    
     |  URLRequired(*args, **kwargs)
     |  
     |  A valid URL is required to make a request.
     |  
     |  Method resolution order:
     |      URLRequired
     |      RequestException
     |      builtins.OSError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args
    
### UnrewindableBodyError
    
<strong>class UnrewindableBodyError(RequestException)</strong>
    
     |  UnrewindableBodyError(*args, **kwargs)
     |  
     |  Requests encountered an error when trying to rewind a body.
     |  
     |  Method resolution order:
     |      UnrewindableBodyError
     |      RequestException
     |      builtins.OSError
     |      builtins.Exception
     |      builtins.BaseException
     |      builtins.object
     |  
     |  Methods inherited from RequestException:
     |  
     |  
#### RequestException.__init__
     |  
     |  __init__(self, *args, **kwargs)
     |      Initialize RequestException with `request` and `response` objects.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from RequestException:
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__reduce__
     |  
     |  __reduce__(...)
     |      Helper for pickle.
     |  
     |  
#### OSError.__str__
     |  
     |  __str__(self, /)
     |      Return str(self).
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from builtins.OSError:
     |  
     |  
#### OSError.__new__
     |  
     |  __new__(*args, **kwargs) from builtins.type
     |      Create and return a new object.  See help(type) for accurate signature.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.OSError:
     |  
     |  characters_written
     |  
     |  errno
     |      POSIX exception code
     |  
     |  filename
     |      exception filename
     |  
     |  filename2
     |      second exception filename
     |  
     |  strerror
     |      exception strerror
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from builtins.BaseException:
     |  
     |  
#### BaseException.__delattr__
     |  
     |  __delattr__(self, name, /)
     |      Implement delattr(self, name).
     |  
     |  
#### BaseException.__getattribute__
     |  
     |  __getattribute__(self, name, /)
     |      Return getattr(self, name).
     |  
     |  
#### BaseException.__repr__
     |  
     |  __repr__(self, /)
     |      Return repr(self).
     |  
     |  
#### BaseException.__setattr__
     |  
     |  __setattr__(self, name, value, /)
     |      Implement setattr(self, name, value).
     |  
     |  
#### BaseException.__setstate__
     |  
     |  __setstate__(...)
     |  
     |  
#### BaseException.with_traceback
     |  
     |  with_traceback(...)
     |      Exception.with_traceback(tb) --
     |      set self.__traceback__ to tb and return self.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from builtins.BaseException:
     |  
     |  __cause__
     |      exception cause
     |  
     |  __context__
     |      exception context
     |  
     |  __dict__
     |  
     |  __suppress_context__
     |  
     |  __traceback__
     |  
     |  args

